src/activities/discussions/TopicGroupSectionRestrictionsEntity.js

```diff
@@ -1,8 +1,75 @@
+import { Actions, Classes, Rels } from '../../hypermedia-constants.js';
 import { Entity } from '../../es6/Entity.js';
+import { performSirenAction } from '../../es6/SirenAction.js';
+import { RestrictedTopicTileEntity } from './RestrictedTopicTileEntity.js';
 
 /**
  * TopicGroupSectionRestrictionsEntity class representation of a D2L Discussion Topic Group Section Restrictions entity.
  */
 export class TopicGroupSectionRestrictionsEntity extends Entity {
-	// TODO: implement
+	isRestrictedThreads() {
+		if (!this._entity) return;
+		return this._entity.hasClass(Classes.discussions.restrictedThreads);
+	}
+	isRestrictedTopic() {
+		if (!this._entity) return;
+		return this._entity.hasClass(Classes.discussions.restrictedTopic);
+	}
+	hasNoGroupsOrSections() {
+		if (!this._entity) return;
+		return this._entity.hasClass(Classes.discussions.noGroupsOrSections);
+	}
+	isUnRestricted() {
+		return !this.isRestrictedThreads() && !this.isRestrictedTopic() && !this.hasNoGroupsOrSections();
```

Comment:
Should a topic be considered restricted if the course has no groups or sections?

Reply:
No it shouldn't, good catch, I was thinking about all the different states, but the UI already checks for noGroupsOrSections separately, so it doesn't make sense to include that here. I'll remove it.